mesh/mesh.cpp

```diff
@@ -3738,28 +3738,58 @@ Mesh::Mesh(Mesh *mesh_array[], int num_pieces)
 }
 
 Mesh::Mesh(Mesh *orig_mesh, int ref_factor, int ref_type)
+{
+   Array<int> ref_factors(orig_mesh->GetNE());
+   ref_factors = ref_factor;
+   CreateRefinedMesh(orig_mesh, ref_factors, ref_type);
+}
+
+Mesh::Mesh(Mesh *orig_mesh, Array<int> &ref_factors, int ref_type)
+{
+   CreateRefinedMesh(orig_mesh, ref_factors, ref_type);
+}
+
+void Mesh::CreateRefinedMesh(Mesh *orig_mesh, Array<int> &ref_factors,
+                             int ref_type)
 {
    Dim = orig_mesh->Dimension();
-   MFEM_VERIFY(ref_factor >= 1, "the refinement factor must be >= 1");
-   MFEM_VERIFY(ref_type == BasisType::ClosedUniform ||
-               ref_type == BasisType::GaussLobatto, "invalid refinement type");
-   MFEM_VERIFY(Dim == 1 || Dim == 2 || Dim == 3,
-               "only implemented for Segment, Quadrilateral and Hexahedron "
-               "elements in 1D/2D/3D");
-   MFEM_VERIFY(orig_mesh->GetNumGeometries(Dim) <= 1,
-               "meshes with mixed elements are not supported");
```

Comment:
Some of these checks should probably be brought back (I think I was the one who deleted them in the first place...)

Reply:
OK, which ones do you want brought back? (I don't know this part of the code so I'm really not sure)

Reply:
For now, all of them (except maybe the `Dim` test, which seems unnecessary)

Reply:
Done.